packages/ui/src/components/GlobalSearch/index.tsx

```diff
@@ -0,0 +1,361 @@
+import { FlashList, ListRenderItem } from '@shopify/flash-list';
+import type * as db from '@tloncorp/shared/db';
+import * as store from '@tloncorp/shared/store';
+import {
+  ChatListItem,
+  ChatListItemData,
+  LoadingSpinner,
+  SectionListHeader,
+  Text,
+  TextInputWithIconAndButton,
+  View,
+  XStack,
+  YStack,
+  getChatKey,
+  getItemType,
+  isSectionHeader,
+  useFilteredChats,
+} from '@tloncorp/ui';
+import { useCallback, useEffect, useMemo, useRef, useState } from 'react';
+import {
+  LayoutChangeEvent,
+  NativeSyntheticEvent,
+  TextInput,
+  TextInputKeyPressEventData,
+} from 'react-native';
+import { getTokenValue } from 'tamagui';
+
+import { useGlobalSearch } from '../../contexts/globalSearch';
+
+export interface GlobalSearchProps {
+  navigateToGroup: (id: string) => void;
+  navigateToChannel: (channel: db.Channel) => void;
+}
+
+export function GlobalSearch({ navigateToGroup, navigateToChannel }: GlobalSearchProps) {
+  const { isOpen, setIsOpen } = useGlobalSearch();
+  const [searchQuery, setSearchQuery] = useState('');
+  const [selectedIndex, setSelectedIndex] = useState(0);
+  const inputRef = useRef<TextInput>(null);
+  const groupsQuery = store.useGroups({});
+  const contactsQuery = store.useContacts();
+
+  const isLoading = groupsQuery.isLoading || contactsQuery.isLoading;
+  const hasError = groupsQuery.error || contactsQuery.error;
+
+  const { data: chats } = store.useCurrentChats({
+    enabled: isOpen,
+  });
+
+  const resolvedChats = useMemo(() => {
+    return {
+      pinned: chats?.pinned ?? [],
+      unpinned: chats?.unpinned ?? [],
+    };
+  }, [chats]);
+
+  const displayData = useFilteredChats({
```

Comment:
is the object literal syntax going to re-run the hook on every render? Should that input be memoized?